### DISCUSSION TITLE
Question. How do I change my username without losing the badges?

### DISCUSSION THREAD

**Discussion by artyomboyko** [community/community#142188] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Good afternoon. Is there any way to change my username without losing the badges I already have?

**Comment by davevad93** [comment, marked as answer]
Hi @blademoon ,

According to GitHub's documentation on [changing username](https://docs.github.com/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username), it shouldn't affect achievements, in fact it doesn't even mention them. GitHub will automatically redirect references to your repositories and commits won't be affected. The only things that would be affected are @mentions using your old username, links to gists that include your old username, marketplace actions and packages. That's what's stated in the docs, I changed the username a while ago and at that time, I had the `Quickdraw` and `Pull Shark` achievements and they haven't disappeared, I don't know about other achievements.

**Reply by artyomboyko** [reply]
Thanks. I gathered my courage and did it))) Honestly, I was really worried about the badges. They didn't come very easily.

**Reply by davevad93** [reply]
You're welcome @artyomboyko 👍 I totally understand, it would be quite annoying to lose all the achievements just for changing the username, but luckily it can be done without any problem.
